src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1154,7 +1152,7 @@ private Map<String, Map<String, Collection<String>>> loadAliases(final ResourceR
 
         log.debug("start alias query: {}", queryString);
         long queryStart = System.nanoTime();
-        final Iterator<Resource> i = resolver.findResources(queryString, "sql");
+        final Iterator<Resource> i = resolver.findResources(queryString, "JCR-SQL2");
```

Comment:
I'd suggest using the Query.JCR_SQL2 constant here instead.

[Query JavaDocs](https://javadoc.io/static/javax.jcr/jcr/2.0/javax/jcr/query/Query.html)

Reply:
Not convinced. We currently do not use javax.jcr imports; likely because Sling in theory supports other backends.

I personally do not buy this argument, but if we go down that road, there's a lot more we could / should change.

Reply:
Pretty weak that you just merging immediately without giving me a chance to respond.   You are already using a JCR specific language there so pretending that a different backend would support that language somehow seems unlikely to me.  Using a string doesn't magically make it more portable.

Reply:
We already optionally import `javax.jcr`, so I don't see what the issue would be with referencing the constant value.

https://github.com/apache/sling-org-apache-sling-resourceresolver/blob/c7bdc4d6a5e360623b0ad3c117f1728e26c12e8e/bnd.bnd#L2

The java compiler will inline string references in the resulting class file anyway.

Reply:
> The java compiler will inline string references in the resulting class file anyway.

Yes, this is true.  I prefer using the Query.JCR_SQL2 constant just in case that language gets deprecated someday (like what happened for the "xpath" and "sql" languages) so that your IDE will warn you about usage of deprecated code in that scenario.